src/control/control.cpp

```diff
@@ -170,7 +170,7 @@ QList<QSharedPointer<ControlDoublePrivate>> ControlDoublePrivate::getAllInstance
 }
 
 // static
-QList<QSharedPointer<ControlDoublePrivate>> ControlDoublePrivate::takeAllInstances() {
+const QList<QSharedPointer<ControlDoublePrivate>> ControlDoublePrivate::takeAllInstances() {
```

Comment:
I think I would prefer the local copy. This should be a no op, because of return value optimization.  
Alternative we can use a qAsConst() as usually. 

The const return value probably prevents the return value optimization, because we can't move from a const. 

That is an interesting question though. I will check.

Reply:
`qAsConst()` doesn't work with return values though. Let's do the local copy then.